lighthouse-core/config/config.js

```diff
@@ -314,19 +314,18 @@ class Config {
   }
 
  /**
-  * Filter out any unrequested items from the config, based on requested top-level aggregations.
-  * @param {!Object} oldConfig Lighthouse config object
-  * @param {!Array<string>} aggregationIDs Id values of aggregations to include
-  * @return {Object} new config
-  */
-  static generateNewConfigOfAggregations(oldConfig, aggregationIDs) {
+   * @param {!Object} config
+   * @param {!Array<string>} categoryIds
+   * @return {!Object}
+   */
+  static generateNewConfigOfCategories(config, categoryIds) {
     // 0. Clone config to avoid mutating it
-    const config = JSON.parse(JSON.stringify(oldConfig));
-    // 1. Filter to just the chosen aggregations
-    config.aggregations = config.aggregations.filter(agg => aggregationIDs.includes(agg.id));
+    config = Object.assign({}, config);
```

Comment:
> I dislike the idea of random parts of config not supporting a consistent feature set a little more than being careful about mutation

yeah, I agree. I was thinking more of future changes to the config format and having to reverify this invariant every time. e.g. we add some object on the audit objects in the future and in that PR we forget to verify that we `Object.assign()` at that level as well, and then (maybe in some future PR after that) the report generator or whatever augments those audit objects, and then we have those extra properties on subsequent uses of the default config in the extension/tests.

Part of that is inevitable due to `require()`'s behavior, and maybe we should only be loading config files with `fs.readFile()`, but regardless it's technical debt which we noted we had when we first wrote `JSON.parse(JSON.stringify())`, so we really should have come up with a better solution before now :S